src/bake/DevServer.zig

```diff
@@ -6018,16 +6114,33 @@ const HmrSocket = struct {
             },
             .set_url => {
                 const pattern = msg[1..];
-                const rbi = s.dev.routeToBundleIndexSlow(pattern) orelse
+                const rbi = s.dev.routeToBundleIndexSlow(pattern) orelse {
+                    if (s.inspector_connection_id > -1) {
+                        if (s.inspector()) |agent| {
+                            var pattern_str = bun.String.init(pattern);
+                            defer pattern_str.deref();
+                            agent.notifyClientNavigated(s.inspector_connection_id, &pattern_str, -1);
+                        }
+                    }
                     return;
+                };
+
                 if (s.active_route.unwrap()) |old| {
                     if (old == rbi) return;
                     s.dev.routeBundlePtr(old).active_viewers -= 1;
                 }
                 s.dev.routeBundlePtr(rbi).active_viewers += 1;
                 s.active_route = rbi.toOptional();
                 var response: [5]u8 = .{MessageId.set_url_response.char()} ++ std.mem.toBytes(rbi.get());
+
                 _ = ws.send(&response, .binary, false, true);
+                if (s.inspector_connection_id > -1) {
```

Comment:
this code is duplicated. just do this first before the branching happens.

Reply:
it's not duplicated. the first version has the route bundle ID. the second one does not.

Reply:
I extracted it into a function so it's clearer why it's not duplicated

Reply:
that approach works, but i should have been more clear this is what i was expecting.

![CleanShot 2025-04-27 at 09 25 39@2x](https://github.com/user-attachments/assets/30532b46-1ada-4e0b-9c8d-e798059bd8ed)